mdtraj/core/selection.py

```diff
@@ -75,7 +76,7 @@ def visit_Name(self, node):
         # all other bare names are taken to be string literals. Thus something
         # like parse_selection('name CA') properly resolves CA as a string
         # literal, not a barename to be loaded from the global scope!
-        return ast.Constant(s=node.id, kind=None)
+        return ast.Constant(value=node.id, kind=None)
```

Comment:
Haha, did this have a named argument `s` in years past?

Also, what's `kind` do ... ? I don't see it in the docs https://docs.python.org/3.12/library/ast.html#ast.Constant

Reply:
The AST documentation is kinda bad.  ```kind``` and `s` are listed in the [grammar tree in the main doc](https://docs.python.org/3/library/ast.html#abstract-grammar), but better explained here: 

> kind is 'u' for strings with a u prefix, and None otherwise, allowing tools to distinguish u"a" from "a".
https://greentreesnakes.readthedocs.io/en/latest/nodes.html#Constant


The s was there for backward compatibility reasons. ```ast.Str(s=something)``` There was a deprecation warning recently: https://github.com/mdtraj/mdtraj/actions/runs/9404499617/job/25903544222#step:6:2330
> A string. The s attribute hold the value. In Python 2, the same type holds unicode strings too.
https://greentreesnakes.readthedocs.io/en/latest/nodes.html#Str